sys/linux/socket_netlink_generic_devlink.txt

```diff
@@ -71,3 +71,61 @@ devlink_port_set {
 	port_handle		devlink_port_handle
 	DEVLINK_ATTR_PORT_TYPE	nlattr[DEVLINK_ATTR_PORT_TYPE, int16[DEVLINK_PORT_TYPE_NOTSET:DEVLINK_PORT_TYPE_IB]]
 } [packed, align_4]
+
+type msghdr_nl_devlink_trap_set msghdr_netlink[netlink_msg_t[genl_devlink_family_id, genlmsghdr_t[DEVLINK_CMD_TRAP_SET], devlink_trap_set]]
+sendmsg$DEVLINK_CMD_TRAP_SET(fd sock_nl_generic, msg ptr[in, msghdr_nl_devlink_trap_set], f flags[send_flags])
+
+devlink_trap_handle {
+	DEVLINK_ATTR_TRAP_NAME	nlattr[DEVLINK_ATTR_TRAP_NAME, string["source_mac_is_multicast"]]
+} [packed, align_4]
+
+devlink_trap_action_handle {
+	DEVLINK_ATTR_TRAP_ACTION	nlattr[DEVLINK_ATTR_TRAP_ACTION, int8[DEVLINK_TRAP_ACTION_DROP:DEVLINK_TRAP_ACTION_TRAP]]
+} [packed, align_4]
+
+devlink_trap_set_arg [
+	action_handle	devlink_trap_action_handle
+] [varlen]
+
+devlink_trap_set {
+	dev_handle	devlink_handle
+	trap_handle	devlink_trap_handle
+	arg		devlink_trap_set_arg
+} [packed, align_4]
+
+type msghdr_nl_devlink_trap_get msghdr_netlink[netlink_msg_t[genl_devlink_family_id, genlmsghdr_t[DEVLINK_CMD_TRAP_GET], devlink_trap_get]]
+sendmsg$DEVLINK_CMD_TRAP_GET(fd sock_nl_generic, msg ptr[in, msghdr_nl_devlink_trap_get], f flags[send_flags])
+
+devlink_trap_get_arg [
+	dev_handle	devlink_handle
+	trap_handle	devlink_trap_handle
+] [varlen]
+
+devlink_trap_get {
```

Comment:
It looks like this devlink_trap_get is unnecessary, we could use devlink_trap_get_arg directly instead of it.